### DISCUSSION TITLE
postgres_fdw integration with db instance hosted on GCP

### DISCUSSION THREAD

**Discussion by admin-siterecon** [supabase/supabase#13821] [Questions]
Hi everyone,

I am trying to evaluate supabase for my enterprise needs. The postgres_fdw extension seems to be a nice start for us. I have setup the foreign tables on supabase. Now I need to connect my GCP DB instance by creating a "server" at supabase DB. The issue is, we follow IP whitelisting as a way to protect unwanted DB accesses. Hence, if I run -
```
CREATE SERVER siterecon_server FOREIGN DATA WRAPPER postgres_fdw OPTIONS (host 'XX.XX.XX.XX', dbname 'my-production-db', port '5432');
```
My DB instance would reject the connections from supabase.

I quickly did a 
```
dig <SUPABASE_DB_HOST>
```

and found out the IP address of the EC2 instance where the DB is hosted. Then I tried connecting my DB and it worked. I was able to query the foreign tables.

Coming to the issues - 
1. What if the IP address of the EC2 instance changes? Is it static for my supabase project?
2. This is not a secure way for us to go if there are other projects hosted on the same EC2 instance.

Kindly advise.

**Comment by admin-siterecon** [comment]
Hey @gregnr @steve-chavez 
Could you please tag any relevant team member who can help me with this query?
Thanks!

**Reply by gregnr** [reply]
This is a good question @admin-siterecon, thanks for reaching out.

@darora any chance you have insight on this?

**Comment by pcnc** [comment]
// Updated at Apr 28 22:25Z for correctness

Hi @admin-siterecon 

Jumping in with some answers

1. We do not co-locate multiple projects at this time. However, this is not a guaranteed commitment, but an artifact of our current architecture. If you need to rely upon it, you should reach out to our Enterprise sales channel and we can work something out (I see that you already have, but stating it here for posterity)
2. The IP address should not be regarded as static. There are a number of both user-initiated and platform-initiated changes that can cause the IP address to be updated. If you require a static IP, we can look into supporting this as part of an Enterprise package.
